cli/main.go

```diff
@@ -292,135 +76,27 @@ func main() {
-
-					eigenpod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
-					core.PanicOnError("failed to connect to eigenpod", err)
-
-					if currentCheckpoint == 0 {
-						if len(sender) > 0 || simulateTransaction {
-							if !noPrompt && !simulateTransaction {
-								core.PanicIfNoConsent(core.StartCheckpointProofConsent())
-							}
-
-							txn, err := core.StartCheckpoint(ctx, eigenpodAddress, sender, chainId, eth, forceCheckpoint, simulateTransaction)
-							core.PanicOnError("failed to start checkpoint", err)
-
-							if !simulateTransaction {
-								color.Green("starting checkpoint: %s.. (waiting for txn to be mined)", txn.Hash().Hex())
-								bind.WaitMined(ctx, eth, txn)
-								color.Green("started checkpoint! txn: %s", txn.Hash().Hex())
-							} else {
-								printProofs([]Transaction{
-									{
-										Type:     "checkpoint_start",
-										To:       txn.To().Hex(),
-										CallData: common.Bytes2Hex(txn.Data()),
-									},
-								})
-
-								return nil
-							}
-
-							newCheckpoint, err := eigenpod.CurrentCheckpointTimestamp(nil)
-							core.PanicOnError("failed to fetch current checkpoint", err)
-
-							currentCheckpoint = newCheckpoint
-						} else {
-							core.PanicOnError("no checkpoint active and no private key provided to start one", errors.New("no checkpoint"))
-						}
-					}
-
-					if isVerbose {
-						color.Green("pod has active checkpoint! checkpoint timestamp: %d", currentCheckpoint)
-					}
-
-					proof, err := core.GenerateCheckpointProof(ctx, eigenpodAddress, eth, chainId, beaconClient)
-					core.PanicOnError("failed to generate checkpoint proof", err)
-
-					jsonString, err := json.Marshal(proof)
-					core.PanicOnError("failed to generate JSON proof data.", err)
-
-					if out != nil {
-						core.WriteOutputToFileOrStdout(jsonString, out)
-					} else if len(sender) != 0 || simulateTransaction {
-						txns, err := core.SubmitCheckpointProof(ctx, sender, eigenpodAddress, chainId, proof, eth, batchSize, noPrompt, simulateTransaction)
-						if simulateTransaction {
-							printableTxns := aMap(txns, func(txn *types.Transaction) Transaction {
-								return Transaction{
-									To:       txn.To().Hex(),
-									CallData: common.Bytes2Hex(txn.Data()),
-									Type:     "checkpoint_proof",
-								}
-							})
-							printProofs(printableTxns)
-						} else {
-							for i, txn := range txns {
-								color.Green("transaction(%d): %s", i, txn.Hash().Hex())
-							}
-						}
-						core.PanicOnError("an error occurred while submitting your checkpoint proofs", err)
-					}
-
-					return nil
+					return commands.CheckpointCommand(commands.TCheckpointCommandArgs{
+						DisableColor:        disableColor,
+						NoPrompt:            noPrompt,
+						SimulateTransaction: len(sender) == 0,
+						BatchSize:           batchSize,
+						ForceCheckpoint:     forceCheckpoint,
+						Node:                node,
+						BeaconNode:          beacon,
```

Comment:
This and the credentials command no longer have the ability to submit txns. You're not passing the `sender` value in here.

Reply:
fixed